### PR TITLE
Update release notes to include 8.7.0

### PR DESCRIPTION
**Pull request by craigtaverner**
Release notes and migration guide from 8.7.0 release ported into main as well as re-generating 8.8.0 release notes. This latter step will be overwritten anyway, multiple times, by more up-to-date regeneration of the 8.8.0 release notes during the release process.

### CODE REVIEW

**Review by abdonpijpelink — APPROVED** [review]
LGTM

**Comment by craigtaverner** [comment]
@elasticmachine run elasticsearch-ci/docs

**Comment by craigtaverner** [comment]
@elasticsearchmachine run elasticsearch-ci/docs

File: docs/reference/migration/migrate_8_7.asciidoc [hunk 1]
```diff
@@ -7,16 +7,39 @@
 This section discusses the changes that you need to be aware of when migrating
 your application to {es} 8.7.
 
-See also <<release-highlights>> and <<es-release-notes>>.
-
-coming::[8.7.0]
-
+See also {ref-bare}/8.7/release-highlights.html[What's new in 8.7] and <<es-release-notes>>.
```
Comment (abdonpijpelink) [inline docs/reference/migration/migrate_8_7.asciidoc:10]:
```suggestion
See also <<release-highlights>> and <<es-release-notes>>.
```

**Review by abdonpijpelink — COMMENTED** [review]
The CI error is because of something I broke in [#92823](https://github.com/elastic/elasticsearch/pull/92823/files#diff-d36c4cd9de9033056522120061bc8d2120e4af4d1b8a62c4c1094e86923df20fR10). I still need to revert that PR. 

This change will allow this PR to pass the CI checks.

**Pull request merged**